app/src/main/java/com/monstarlab/core/usecases/FlowUseCase.kt

```diff
@@ -0,0 +1,14 @@
+package com.monstarlab.core.usecases
+
+import com.monstarlab.arch.extensions.UseCaseResult
+import com.monstarlab.arch.extensions.useCaseFlow
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flowOn
+
+abstract class FlowUseCase<in P, R>(private val coroutineDispatcher: CoroutineDispatcher) {
+    operator fun invoke(parameters: P): Flow<UseCaseResult<R>> = useCaseFlow { execute(parameters) }
+        .flowOn(coroutineDispatcher)
+
+    protected abstract suspend fun execute(parameters: P): R
+}
```

Comment:
I think we should try another angle on this. We tried to avoid having abstract *UseCase classes for more flexibility on the Input/Output part. 

Can we do this in another way via extensions?

Reply:
got it. it's possible.

Reply:
implementation changed